internal/database/arango/connection.go

```diff
@@ -0,0 +1,151 @@
+
+type ArangoDB interface {
+	Database(ctx context.Context, dbName string) (arangodb.Database, error)
+	CreateCollection(ctx context.Context, dbName string, colName string) (arangodb.Collection, error)
+	Collection(ctx context.Context, dbName string, colName string) (arangodb.Collection, error)
+	AsyncDatabase(ctx context.Context, jobID string, dbName string) (arangodb.Database, error)
+	AsyncJobResult(ctx context.Context, dbName string) (arangodb.Database, string, error)
+}
+
+type arangoDB struct {
+	connection connection.Connection
+}
+
+func NewArangoDB(ctx context.Context, conf *config.ArangoConfig) (ArangoDB, error) {
+	connStrs, err := config.GetArangoStrings(conf)
+	if err != nil {
+		return nil, err
+	}
+	endpoint := connection.NewRoundRobinEndpoints(connStrs)
+	conn := connection.NewHttp2Connection(connection.DefaultHTTP2ConfigurationWrapper(endpoint /*InsecureSkipVerify*/, conf.InsecureSkipVerify))
+
+	auth := connection.NewBasicAuth(conf.User, conf.Pass)
+	err = conn.SetAuthentication(auth)
+	if err != nil {
+		return nil, err
+	}
+
+	return &arangoDB{
+		connection: conn,
+	}, nil
+}
+
+func (a *arangoDB) syncClient() arangodb.Client {
+	return arangodb.NewClient(a.connection)
+}
+
+func (a *arangoDB) asyncClient() arangodb.Client {
+	conn := connection.NewConnectionAsyncWrapper(a.connection)
+	return arangodb.NewClient(conn)
+}
+
+func (a *arangoDB) Database(ctx context.Context, dbName string) (arangodb.Database, error) {
+	timeoutCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	client := a.syncClient()
+	dbExists, err := client.DatabaseExists(timeoutCtx, dbName)
+	if err != nil {
+		return nil, err
+	}
+	if !dbExists {
+		return nil, fmt.Errorf("database %s does not exist", dbName)
+	}
+
+	db, err := client.Database(timeoutCtx, dbName)
+	if err != nil {
+		return nil, err
+	}
+
+	return db, nil
+}
+
+func (a *arangoDB) CreateCollection(ctx context.Context, dbName string, colName string) (arangodb.Collection, error) {
+	db, err := a.Database(ctx, dbName)
+	if err != nil {
+		return nil, err
+	}
+
+	exists, err := db.CollectionExists(ctx, colName)
+	if err != nil {
+		return nil, err
+	}
+	if exists {
+		return nil, fmt.Errorf("collection %s already exists", colName)
+	}
+	options := arangodb.CreateCollectionOptions{}
+	properties := arangodb.CreateCollectionProperties{}
+
+	collection, err := db.CreateCollectionWithOptions(ctx, colName, &properties, &options)
+	if err != nil {
+		return nil, err
+	}
+	return collection, nil
+}
+
+func (a *arangoDB) Collection(ctx context.Context, dbName string, colName string) (arangodb.Collection, error) {
+	db, err := a.Database(ctx, dbName)
+	if err != nil {
+		return nil, err
+	}
+
+	options := arangodb.GetCollectionOptions{}
+
+	collection, err := db.GetCollection(ctx, colName, &options)
+	if err != nil {
+		return nil, err
+	}
+
+	return collection, nil
+}
+
+func (a *arangoDB) AsyncJobResult(ctx context.Context, dbName string) (arangodb.Database, string, error) {
```

Comment:
on this function, `arangodb.Database` always returns `nil`!
Please remove this output parameter like this:
```
func (a *arangoDB) AsyncJobResult(ctx context.Context, dbName string) (string, error)
```